shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm

```diff
@@ -287,4 +291,19 @@ - (NSDictionary*)clipboardHasStrings {
   return @{@"value" : @(hasStrings)};
 }
 
+- (BOOL)isLiveTextInputAvailable {
+  return [[self textField] canPerformAction:@selector(captureTextFromCamera:) withSender:nil];
```

Comment:
We can create and release the `UITextField` object in this call only once, and save the result of `canPerformAction:@selector(captureTextFromCamera:)` in a local bool. 
This way we only create the text field once and releases it as early as possible.

Example:
```objc
- (BOOL)isLiveTextInputAvailable {
  if (_isLiveTextInputAvailable) {
    return [_isLiveTextInputAvailable boolValue];
  }
  UITextField *textField = [[UITextField alloc] init];
  _isLiveTextInputAvailable = @([textField canPerformAction:@selector(captureTextFromCamera:) withSender:nil]);
  [textField release];
  return [_isLiveTextInputAvailable boolValue];
}

```

Reply:
The value `canPerformAction:@selector(captureTextFromCamera:)` returns may change when the current system settings change (most notably, when camera is disabled for the app by the user).

Reply:
Emm I think we can't save the it as a local value due to: https://github.com/flutter/engine/pull/29229#discussion_r886245273

Reply:
I see. Usually those settings will force the app to restart, but if user can change the setting value and the app won't restart, my suggestion won't work.